src/components/input/textarea-input.tsx

```diff
@@ -0,0 +1,90 @@
+import { ChangeEventHandler, ReactNode, useCallback } from 'react';
+import NumberFormat from 'react-number-format';
+import styles from './textarea-input.module.scss';
+import { CrossIcon } from '../icons';
+import classNames from 'classnames';
+
+interface InputProps {
+  autoFocus?: boolean;
+  multiline?: false;
+  type?: 'text' | 'number';
+}
+
+interface MultilineInputProps {
+  type?: 'text';
+  multiline?: true;
+}
+
+type Props = {
+  id: string;
+  helperText?: string;
+  error?: boolean;
+  placeholder?: string;
+  value: string;
+  onChange: ChangeEventHandler<HTMLInputElement>;
+} & (InputProps | MultilineInputProps);
+
+const Input = ({ type = 'text', multiline, ...componentProps }: Props) => {
+  const CustomNumberInput = useCallback((props: any) => {
+    return (
+      <div className={styles.input}>
+        <input {...props} />
+      </div>
+    );
+  }, []);
+
+  if (multiline) {
```

Comment:
Hmm I'm a bit confused. Can we extract the textarea back into its own `<Textarea />` component? I don't see the need to combine it with other things. Top component libs like shadcn, headless UI, react-aria keeps it separate. Also the name `<TextAreaInput />` can be misleading as it almost sounds like it should always render a textarea. I wouldn't naturally reach for it for a standard input.

Reply:
It made sense for me in our case because it's aligned with the designs this way. We have three different inputs in the designs. It feels to me that 1 & 2 (not implemented yet) make sense to share the same component, while 3 can be a separate one.

### 1. Deposit/Stake/Unstake/...etc. modals ([design](https://www.figma.com/design/NG6QClnVrvPRZ5gJ3s0zen/%E2%9F%81--Api3-DAO---Branding-Update?node-id=2373-67360&m=dev))
- Called `FormField` in designs
- Implemented. Uses Autosize

<img width="216" alt="image" src="https://github.com/user-attachments/assets/9b1c6fe1-9180-4f88-a18d-e2231d9e4a10">

### 2. Delegate modal ([design](https://www.figma.com/design/NG6QClnVrvPRZ5gJ3s0zen/%E2%9F%81--Api3-DAO---Branding-Update?node-id=8272-117999&m=dev))
- Called `FormField` in designs (but different group)
- Not yet implemented. Simple, no Autosize

<img width="337" alt="image" src="https://github.com/user-attachments/assets/9b89feab-15c4-44f2-b7da-8e9137db92ed">

### 3. New Proposal modal ([design](https://www.figma.com/design/BhnOcskTIGPZ18JkhbCuNQ/%E2%9F%81--Api3-DS---Branding-Update?node-id=890-6963&m=dev))
- Called `Input` in designs
- Implemented in `TextareaInput.tsx`

<img width="330" alt="image" src="https://github.com/user-attachments/assets/6a320771-c826-413e-9c28-e3070f8099ad">

Reply:
>  I don't see the need to combine it with other things. Top component libs like shadcn, headless UI, react-aria keeps it separate.

I actually see the point. There are two arguments I have for this approach:
A) We are aligned with the designs. Both `input` and `textarea` are similar and part of the same component in Figma, so they could share the styling in code as well

B) We don't need to be as modular as those libraries since our use case is very specific

Reply:
Ok, let's name this `<Input />` then.